### PR TITLE
1. Added support for StreamAtChange.

### PR DESCRIPTION
**Pull request by skumar7322**
2. Added p4d 19.1 to run test. streamlog command is supported after this version.

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/p4/workspace/ManualWorkspaceImpl.java [hunk 1]
```diff
@@ -233,6 +234,12 @@ public IClient setClient(IOptionsServer connection, String user) throws Exceptio
 		}
 		iclient.setStream(streamFullName);
 
+		String streamAtChange = getSpec().getStreamAtChange();
+		if(StringUtils.isNotEmpty(streamAtChange)){
+			streamAtChange = getExpand().format(streamAtChange,true);
+			iclient.setStreamAtChange(Integer.parseInt(streamAtChange));
```
Comment (p4paul) [inline src/main/java/org/jenkinsci/plugins/p4/workspace/ManualWorkspaceImpl.java:240]: streamAtChange could be an automatic label (e.g. String).  If we add 64bit long support to P4Java than change will be a long and not int (but that's for the future).  If using Integer.parseInt you may want to catch the Runtime NumberFormatException.

File: src/main/java/org/jenkinsci/plugins/p4/workspace/WorkspaceSpec.java [hunk 2]
```diff
@@ -106,7 +114,7 @@ public WorkspaceSpec(String view, String changeView) {
 	public WorkspaceSpec(boolean allwrite, boolean clobber, boolean compress,
 	                     boolean locked, boolean modtime, boolean rmdir, String streamName,
 	                     String line, String view) {
-		this(allwrite, clobber, compress, locked, modtime, rmdir, streamName, line, view, null, null, null, true);
+		this(allwrite, clobber, compress, locked, modtime, rmdir, streamName, "",line, view, null, null, null, true);
```
Comment (p4paul) [inline src/main/java/org/jenkinsci/plugins/p4/workspace/WorkspaceSpec.java:117]: Might need to preserve the Deprecated method and create a new Deprecated method signature (to allow for all the old versions of pipeline code)

File: src/test/java/org/jenkinsci/plugins/p4/DefaultEnvironment.java [hunk 3]
```diff
@@ -156,7 +158,7 @@ private ManualWorkspaceImpl createWorkspace(String path) {
 		String stream = null;
 		String line = "LOCAL";
 		String view = "\"" + path + "\"" + " //" + client + "/" + filename;
-		WorkspaceSpec spec = new WorkspaceSpec(true, true, false, false, false, false, stream, line, view, null, null, null, true);
+		WorkspaceSpec spec = new WorkspaceSpec(true, true, false, false, false, false, stream, null,line, view, null, null, null, true);
```
Comment (p4paul) [inline src/test/java/org/jenkinsci/plugins/p4/DefaultEnvironment.java:161]: You wouldn't need to change the tests if you added a Deprecated method signature to match the old pipeline api.

File: src/main/java/org/jenkinsci/plugins/p4/workspace/WorkspaceSpec.java (hunk 2 above)
Comment (skumar7322) [reply src/main/java/org/jenkinsci/plugins/p4/workspace/WorkspaceSpec.java:117]: @p4paul , streamAtChange is added to '@DataBoundConstructor' constructor. This deprecated method calls this constructor, hence adding empty streamAtChange is required. Also, the constructor is used in the Test cases, these test require this change.

**Pull request merged**